drivers/sensor/wsen_hids/hids.c

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (c) 2022 Würth Elektronik eiSos GmbH & Co. KG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT we_wsen_hids
+
+#include <string.h>
+
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/logging/log.h>
+
+#include "hids.h"
+
+LOG_MODULE_REGISTER(HIDS, CONFIG_SENSOR_LOG_LEVEL);
+
+struct str2odr {
+	const char *str;
+	HIDS_outputDataRate_t odr;
+};
+
+/* Maps ODR configuration string (CONFIG_HIDS_ODR) to enum value used by HIDS driver. */
+static const struct str2odr hids_odr_string_list[] = {
+	{ "1", HIDS_odr1Hz },
+	{ "7", HIDS_odr7Hz },
+	{ "12.5", HIDS_odr12_5Hz },
+};
+
+/*
+ * List of supported output data rates (sensor_value struct, input to
+ * sensor_attr_set()). Index into this list is used as argument for
+ * HIDS_setOutputDataRate().
+ */
+static const struct sensor_value hids_odr_list[] = { { .val1 = 0, .val2 = 0 },
+						     { .val1 = 1, .val2 = 0 },
+						     { .val1 = 7, .val2 = 0 },
+						     { .val1 = 12, .val2 = 5 * 100000 } };
+
+static int hids_sample_fetch(const struct device *dev, enum sensor_channel channel)
+{
+	struct hids_data *data = dev->data;
+	int16_t rawHumidity;
+	int16_t rawTemp;
+
+	__ASSERT_NO_MSG(channel == SENSOR_CHAN_ALL);
+
+	if (HIDS_getRawValues(&data->sensor_interface, &rawHumidity, &rawTemp) != WE_SUCCESS) {
+		LOG_ERR("Failed to fetch data sample.");
+		return -EIO;
+	}
+
+	if (WE_SUCCESS !=
+	    HIDS_convertHumidity_uint16(&data->sensor_interface, rawHumidity, &data->humidity)) {
+		LOG_ERR("Failed to convert humidity sample.");
+		return -EIO;
+	}
+	if (WE_SUCCESS !=
```

Comment:
it would be clearer to the reader if theres an extra line here to provide separation

Reply:
Done.